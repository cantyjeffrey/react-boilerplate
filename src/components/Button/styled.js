// @flow

import styled from 'styled-components';

export const Root = styled.button`
  color: #fff;
  background-color: blue;
  border: none;
  border-radius: 3px;
  box-shadow: none;
  margin: 0;
  line-height: 1;
  font-weight: 500;
  letter-spacing: 0.02em;
  padding: 1rem;
`;

export default {};
