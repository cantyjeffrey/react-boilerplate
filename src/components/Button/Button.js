/* eslint-disable react/prop-types */
// @flow

import React from 'react';
import { Root } from './styled';

const Button = ({ children }) => <Root data-testid="Button">{children}</Root>;

export default Button;
