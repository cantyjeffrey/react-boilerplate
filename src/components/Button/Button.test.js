// @flow

import { mount, shallow } from 'enzyme';
import 'jest-styled-components';
import puppeteer from 'puppeteer';
import React from 'react';
import Button from './Button';

const PROTOCOL = process.env.HTTPS === 'true' ? 'https' : 'http';
const PORT = parseInt(process.env.PORT, 10) || 3000;
const HOST = process.env.HOST || '0.0.0.0';
const URL = `${PROTOCOL}://${HOST}:${PORT}`;

describe('<Button />', () => {
  let browser;

  beforeAll(async () => {
    browser = await puppeteer.launch();
  });

  it('should match existing snapshot', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match existing image snapshot', async () => {
    const page = await browser.newPage();
    await page.setViewport({ width: 1366, height: 768 });
    await page.goto(`${URL}`);
    const component = await page.$('[data-testid="Button"]');
    const image = await component.screenshot();

    expect(image).toMatchImageSnapshot({
      customSnapshotIdentifier: 'Button',
    });
  });

  xit('renders a button with correct copy', () => {
    const wrapper = mount(<Button>Super Awesome Button</Button>);
    const button = <button>Super Awesome Button</button>;
    expect(wrapper).toContainReact(button);
  });

  it('renders with correct styles', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toHaveStyleRule('background-color', '#000');
  });

  afterAll(async () => {
    await browser.close();
  });
});
