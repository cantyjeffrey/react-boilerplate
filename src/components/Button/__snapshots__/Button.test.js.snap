// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match existing snapshot 1`] = `
<styled__Root
  data-testid="Button"
/>
`;
